packages/core/src/dialects/sqlite/query-interface.ts

```diff
@@ -1,28 +1,127 @@
+import { withSqliteForeignKeysOff } from './sqlite-utils';
+
+export interface SQLiteColumnDescription extends ColumnDescription {
+  unique?: boolean;
+  references?: {
+    table: string,
+    key: string,
+  };
+}
+
+export type SQLiteColumnsDescription = Record<string, SQLiteColumnDescription>;
 
-/**
- * Temporary class to ease the TypeScript migration
- */
-export class SqliteQueryInterfaceTypeScript extends AbstractQueryInterface {
+export class SqliteQueryInterface extends AbstractQueryInterface {
   readonly sequelize: Sequelize;
   readonly queryGenerator: SqliteQueryGenerator;
-
-  constructor(sequelize: Sequelize, queryGenerator: SqliteQueryGenerator) {
-    super(sequelize, queryGenerator);
+  readonly #internalQueryInterface: AbstractQueryInterfaceInternal;
+
+  constructor(
+    sequelize: Sequelize,
+    queryGenerator: SqliteQueryGenerator,
+    internalQueryInterface?: AbstractQueryInterfaceInternal,
+  ) {
+    super(sequelize, queryGenerator, internalQueryInterface);
     this.sequelize = sequelize;
     this.queryGenerator = queryGenerator;
+    this.#internalQueryInterface = internalQueryInterface ?? new AbstractQueryInterfaceInternal(sequelize, queryGenerator);
+  }
+
+  async dropAllTables(options?: QiDropAllTablesOptions) {
+    const tableNames = await this.showAllTables({ ...options });
+    await withSqliteForeignKeysOff(this.sequelize, { ...options }, async () => {
+      await this._dropAllTables(tableNames, options?.skip || [], options);
+    });
+  }
+
+  async describeTable(tableName: TableNameOrModel, options?: DescribeTableOptions): Promise<SQLiteColumnsDescription> {
+    const table = this.queryGenerator.extractTableDetails(tableName);
+
+    if (typeof options === 'string') {
+      noSchemaParameter();
+      table.schema = options;
+    }
+
+    if (typeof options === 'object' && options !== null) {
+      if (options.schema) {
+        noSchemaParameter();
+        table.schema = options.schema;
+      }
+
+      if (options.schemaDelimiter) {
+        noSchemaDelimiterParameter();
+        table.delimiter = options.schemaDelimiter;
+      }
+    }
+
+    const sql = this.queryGenerator.describeTableQuery(table);
+    try {
+      const data = await this.sequelize.queryRaw(sql, { ...options, type: QueryTypes.DESCRIBE }) as SQLiteColumnsDescription;
+      /*
+       * If no data is returned from the query, then the table name may be wrong.
+       * Query generators that use information_schema for retrieving table info will just return an empty result set,
+       * it will not throw an error like built-ins do (e.g. DESCRIBE on MySql).
+       */
+      if (isEmpty(data)) {
+        throw new Error(`No description found for table ${table.tableName}${table.schema ? ` in schema ${table.schema}` : ''}. Check the table name and schema; remember, they _are_ case sensitive.`);
+      }
+
+      const indexes = await this.showIndex(tableName, { ...options });
```

Comment:
This is not caused by your PR, but why would we include these properties in sqlite, but not the other dialects? The different dialects should be aligned, and in my opinion sqlite should align itself with the other dialects

Could you add a TODO?

Reply:
The reason for this is SQLite's limited ALTER TABLE function and it needs unique and foreign keys in the `ColumnDescription` so that they aren't overridden or removed. I've added comments in the code.

Reply:
Further comments added now.
@ephys LMK if you are happy with these.